hal/network/ncp/cellular/network_config_db.cpp

```diff
@@ -28,23 +29,46 @@ namespace {
 struct NetworkConfig {
     const char* mcc;
     const char* mnc;
+    const char *iccid_prefix;
     const char* apn;
     const char* user;
     const char* password;
 };
 
 const NetworkConfig NETWORK_CONFIG[] = {
-    { "214", "07", "spark.telefonica.com", "", "" }, // Telefonica
-    { "310", "410", "10569.mcs", "", "" }, // Kore/AT&T
-    { "204", "04", "vfd1.korem2m.com", "kore", "kore" } // Kore/Vodafone
+    { "214", "07", "", "spark.telefonica.com", "", "" }, // Telefonica
+    { "310", "410", "", "10569.mcs", "", "" }, // Kore/AT&T
+    { "204", "04", "", "vfd1.korem2m.com", "kore", "kore" } // Kore/Vodafone
+};
+
+const NetworkConfig NETWORK_CONFIG_ICCID[] = {
```

Comment:
Nit: These could potentially be added just to `NETWORK_CONFIG` same way with MCC/MNC just set to null or empty, or we can even add a separate structure with just ICCID and apn/user/password.